backend/service/chaos/experimentation/terminator/terminator.go

```diff
@@ -2,49 +2,119 @@ package terminator
+	if err != nil {
+		m.Run(context.Background())
+	}
+
+	return m, nil
 }
 
-// TODO(snowp): Remove this once we have a proper service object that we can create.
-func NewTestMonitor(store experimentstore.Storer, enabledConfigTypes []string, criterias []TerminationCriteria, log *zap.SugaredLogger, stats tally.Scope) Monitor {
-	return &monitor{
-		store:                      store,
-		enabledConfigTypes:         enabledConfigTypes,
-		criterias:                  criterias,
-		outerLoopInterval:          1,
-		perExperimentCheckInterval: 1,
-		log:                        log,
-		activeMonitoringRoutines:   trackingGauge{gauge: stats.Gauge("active_monitoring_routines")},
-		criteriaEvaluationSuccess:  stats.Counter("criteria_success"),
-		criteriaEvaluationFailure:  stats.Counter("criteria_failure"),
-		terminationCount:           stats.Counter("terminations"),
-		marshallingErrors:          stats.Counter("unpack_error"),
+func NewMonitor(cfg *any.Any, logger *zap.Logger, scope tally.Scope) (*Monitor, error) {
+	typedConfig := &terminatorv1.Config{}
+	err := ptypes.UnmarshalAny(cfg, typedConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	store, ok := service.Registry[experimentstore.Name]
+	if !ok {
+		return nil, errors.New("could not find experiment store service")
+	}
+
+	storer, ok := store.(experimentstore.Storer)
+	if !ok {
+		return nil, errors.New("service was not the correct type")
+	}
+
+	terminationCriteria := map[string][]TerminationCriteria{}
+
+	for configType, perConfigTypeConfig := range typedConfig.PerConfigTypeConfiguration {
+		perConfigCriterias := []TerminationCriteria{}
+
+		for _, c := range perConfigTypeConfig.TerminationCriteria {
+			factory, ok := CriteriaFactories[c.TypeUrl]
+			if !ok {
+				return nil, fmt.Errorf("terminator module configured with unknown criteria '%s'", c.TypeUrl)
+			}
+
+			criteria, err := factory.Create(c)
+			if err != nil {
+				return nil, fmt.Errorf("failed to create termination criteria '%s': %s", c.TypeUrl, err)
+			}
+
+			perConfigCriterias = append(perConfigCriterias, criteria)
+		}
+
+		terminationCriteria[configType] = perConfigCriterias
+	}
+
+	outerLoopInterval, err := ptypes.Duration(typedConfig.OuterLoopInterval)
+	if err != nil {
+		return nil, err
+	}
+
+	perExperimentCheckInterval, err := ptypes.Duration(typedConfig.PerExperimentCheckInterval)
+	if err != nil {
+		return nil, err
 	}
+
+	return &Monitor{
+		store:                        storer,
+		terminationCriteriaByTypeUrl: terminationCriteria,
+		outerLoopInterval:            outerLoopInterval,
+		perExperimentCheckInterval:   perExperimentCheckInterval,
+		log:                          logger.Sugar(),
+		activeMonitoringRoutines:     trackingGauge{gauge: scope.Gauge("active_monitoring_routines")},
+		criteriaEvaluationSuccess:    scope.Counter("criteria_success"),
+		criteriaEvaluationFailure:    scope.Counter("criteria_failure"),
+		terminationCount:             scope.Counter("terminations"),
```

Comment:
feels weird that some of then have `Count` suffix and some of them don't - not a blocker

Reply:
good point, ill make these consistent